release/models/flex-algo/openconfig-flexalgo.yang

```diff
@@ -0,0 +1,397 @@
+module openconfig-flexalgo {
+  yang-version 1.1;
+  namespace "http://openconfig.net/yang/flexalgo";
+  prefix "oc-flexalgo";
+
+  import openconfig-network-instance {
+    prefix "oc-ni";
+  }
+  import openconfig-extensions {
+    prefix oc-ext;
+  }
+  import openconfig-interfaces {
+    prefix oc-if;
+  }
+  oc-ext:openconfig-version "0.1.0";
+
+  organization
+    "OpenConfig working group";
+  contact
+    "OpenConfig working group
+    www.openconfig.net ";
+  description
+    "This YANG module defines the configurational and operational
+    parameters related to IGP Flex-Algorithms";
+
+  revision 2021-12-20 {
+    description
+      "Initial revision of flexalgo module";
+    reference
+      "0.1.0";
+  }
+
+  /* Types definitions */
+
+  typedef flex-algo-metric-type {
+    type enumeration {
+      enum IGP {
+        value 0;
+        description
+          "IGP metric to be used during the calculation.";
+      }
+      enum MIN_DELAY {
+        value 1;
+        description
+          "Min Unidirectional Link Delay as defined in [RFC7810] to
+           be used during the calculation.";
+      }
+      enum TE_METRIC {
+        value 2;
+        description
+          "TE default metric as defined in [RFC5305] to be used
+           during the calculation.";
+      }
+    }
+    description
+      "Type of metric to be used during a Flex Algo path
+       computation";
+  }
+
+  typedef flex-algo-id {
+    type uint8 {
+      range "128..255";
+    }
+    description
+      "Flex-Algorithm identifies which algorithm should be used during
+       path computation";
+  }
+
+  typedef flex-algo-calc-type {
+    type uint8 {
+      range "0..127";
+    }
+    default "0";
+    description
+      "Calculation type value ranges from 0 to 127 both inclusive
+       from the IGP Algorithm Types registry defined under
+       Interior Gateway Protocol (IGP)
+       Parameters IANA registries.
+       If the required calculation type is Shortest Path First,
+       the value 0 SHOULD appear in this field.";
+  }
+
+  grouping algorithm-attributes {
+    description
+      "All of the possible attributes for an flexible algorithm
+       definition";
+    leaf name {
+      type string;
+      description
+        "Flexible Algorithm name";
+    }
+    leaf flex-algo-id {
+      type flex-algo-id;
+      mandatory true;
+      description
+        "Flex Algorithm Identifier";
+    }
+    leaf flex-algo-description {
+      type string;
+      description
+        "Description of this flex-algo";
```

Comment:
This was suggested by @gvandeve so that users can describe the purpose of the flex-algo. Like `low latency paths` or `exclude region x` something like that...